kratos/processes/split_internal_interfaces_process.h

```diff
@@ -201,10 +200,11 @@ class /*KRATOS_API(KRATOS_CORE)*/  SplitInternalInterfaceProcess : public Proces
             for(auto& rNode : geom)
                 ids_on_interface.insert(rNode.Id());
         }
+
         //create duplicated nodes list
         std::size_t max_node_id = 0;
-        if(mrModelPart.GetRootModelPart().Nodes().size() != 0) {
-            max_node_id = (mrModelPart.GetRootModelPart().Nodes().end()-1)->Id() + 1;
+        for(auto& rNode : mrModelPart.Nodes()) {
```

Comment:
nodes is a pointervectorset so it does guarantee to be ordered

Reply:
Did we remove push_back from the API?

Reply:
well, if you look how it is done, it should provide that guarantee.

however after doing a push_back i would call a sort, "just in case"

Reply:
I checked the implementation of 'push_back' and it does not guarantee sorting. You would call a sort after a push_back, but maybe other developers will not. So we need to decide wheter we sort in our process (expensive if needed but very cheap very ofte when it's not needed) or we loop once over the nodes/conditions (constant medium cost). Both options are good because in principle this process only has a ExecuteInitialize, so it is not run many times in principle.
Your call @RiccardoRossi .